src/System.Net.Security/tests/FunctionalTests/KerberosTest.cs

```diff
@@ -0,0 +1,474 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net.Test.Common;
+using System.Security.Authentication;
+using System.Security.Principal;
+using System.Text;
+using System.Threading.Tasks;
+
+using Xunit;
+
+namespace System.Net.Security.Tests
+{
+    public class KDCSetup : IDisposable
+    {
+        private const string Krb5ConfigFile = "/etc/krb5.conf";
+        private const string KDestroyCmd = "kdestroy";
+        private const string SudoCommand = "sudo";
+        private const string ScriptName = "setup-kdc.sh";
+        private const string ScriptUninstallArgs = "--uninstall --yes";
+        private const string PasswordArgFormat = "--password {0}";
+        private readonly bool _isKrbInstalled ;
+        public readonly string password;
+
+        public KDCSetup()
+        {
+            _isKrbInstalled = File.Exists(Krb5ConfigFile);
+            if (!_isKrbInstalled)
+            {
+                password = Guid.NewGuid().ToString("N");
+                int exitCode = RunSetupScript(string.Format(PasswordArgFormat, password));
+                if (exitCode != 0)
+                {
+                    Dispose();
+                    Assert.True(false, "KDC setup failure");
+                }
+            }
+            else
+            {
+                password = TestConfiguration.DefaultPassword;
+            }
+        }
+
+        public void Dispose()
+        {
+            if (!_isKrbInstalled)
+            {
+                RunSetupScript(ScriptUninstallArgs);
+            }
+        }
+
+        // checks for avilability of Kerberos related infrastructure
+        // on the host. Returns true available, false otherwise
+        public bool CheckAndInitializeKerberos()
+        {
+            if (_isKrbInstalled)
+            {
+                // Clear the credentials
+                var startInfo = new ProcessStartInfo(KDestroyCmd);
```

Comment:
Actually, the name of the command is ```kdestroy```. It flushes the credentials from the system cache. Nothing is being destroyed. Without clearing the credentials, some of the tests can fail (or become irrelevant). At the most, this can cause inconvenience - a website might ask the user for credentials again.